webui/react/src/stores/workspaces.tsx

```diff
@@ -34,13 +35,25 @@ class WorkspaceStore extends PollingStore {
     });
   });
 
-  public getWorkspace(id?: number): Observable<Loadable<Workspace>> {
-    return this.workspaces.select((loadable) => {
-      return Loadable.quickMatch(loadable, NotLoaded, (workspaces) => {
-        const workspace = workspaces.find((workspace) => workspace.id === id);
-        return workspace ? Loaded(workspace) : NotLoaded;
-      });
+  public readonly mutables = this.#loadableWorkspaces.select((loadable) => {
+    return Loadable.quickMatch(loadable, NotLoaded, (workspaces) => {
+      return Loaded(
+        workspaces
+          .filter((workspace) => !workspace.immutable)
+          .sort((a, b) => alphaNumericSorter(a.name, b.name)),
+      );
     });
+  });
+
+  public getWorkspace(id: Loadable<number>): Observable<Loadable<Workspace | null>> {
```

Comment:
I recommend making this `id: number | Loadable<number>` so you don't have to unnecessarily wrap the id in cases where it's not coming from another API call.